### ISSUE TITLE
async custom hook not working

### ISSUE THREAD

**Issue by KamikazeSloth** [testing-library/react-hooks-testing-library#417]
This is my first time posting a question, let me know if I should edit my question to make it better and more readable if it's hard to follow. 

I'm trying to make my test work with my custom hook with "react hooks testing library". I've looked through old threads, asked collegues and still cannot make this test work with my async custom hook.

**So the custom hook I have is like this, a custom hook with an api-call in it:** 

```js
export const useGetPolicies = () => {
  const { getTokenSilently } = useAuth0()

  const getPolicies = async (customerId, source) => {
    const token = await getTokenSilently()
    try {
      const response = await axios({
        url: constants.apiCalls.policiesUrl,
        method: 'GET',
        withCredentials: 'include',
        headers: {
          'Authorization': `Bearer ${token}`
        },
        params: {
          customerId: customerId
        }, cancelToken: source.token
      })
      return response
    } catch (error) {
      if (axios.isCancel(error)) {
        console.log(error.message)
      } else {
        return error
      }
    }
  }
  return {
    getPolicies
  }
}
```

**The test I have is like this, it runs the custom hook:** 

```js
import { renderHook, act } from '@testing-library/react-hooks'
import { useGetPolicies } from '../../utils/apiCalls'
import { useAuth0 } from '../../react-auth0-spa';

jest.mock('../../react-auth0-spa');

test("should handleIncrementAsync", async () => {

jest.setTimeout(15000)

 useAuth0.mockReturnValue({
     getTokenSilently: jest.fn()
  })
  const { result, waitForNextUpdate } = renderHook(() => useGetPolicies());
  result.current.getPolicies()
  await waitForNextUpdate()
});
```

**The error I receive is:** 

"Timeout - Async callback was not invoked within the 15000ms timeout specified by jest.setTimeout.Timeout - Async callback was not invoked within the 15000ms timeout specified by jest.setTimeout.Error:"

**My package versions are:**

    "react": "^16.13.1",
    "@testing-library/react-hooks": "^1.1.0",

Why won't my test work? I know the api call works, I will mock it in my test later but first I want the custom hook to work in general in my test. I appreciate any help :)

**Comment by mpeyper** [comment]
Hi @KamikazeSloth,

I'm going to simplify your hook in order to remove the irrelevant parts to what's causing your issue:

```js
export const useGetPolicies = () => {
  const { getTokenSilently } = useAuth0()

  const getPolicies = async (customerId, source) => {
    const token = await getTokenSilently()
    try {
      const response = await apiCall()
      return response
    } catch (error) {
      return error
    }
  }
  return {
    getPolicies
  }
}
```

The reason `waitForNextUpdate` is timing out is that there are no updates to trigger the hook to rerender after the api call promise has resolved.  _Usually_, the result of the call is set into some kind of state which triggers the render and the allows `waitForNextUpdate` to resolve.

```js
export const useGetPolicies = () => {
  const { getTokenSilently } = useAuth0()
  const [policies, setPolicies] = useState()

  const getPolicies = async (customerId, source) => {
    const token = await getTokenSilently()
    try {
      const response = await apiCall()
      setPolicies(response.data)
      return response
    } catch (error) {
      return error // do you really want to return an error here?
    }
  }
  return {
    getPolicies,
    policies
  }
}
```

Of course, you use case may vary, so I'll leave it up to you to decide how to appropriately use the result of the api call, but if there is no intention to update some state or trigger an effects _at all_ than I would suggest you don't actually want a hook for this and it's the wrong tool for your problem.

Let me know how you go.

**Comment by mpeyper** [comment]
Closing this due to inactivity.  Please let me know if you are still having issues.
